pkg/bucketeer/event/processor.go

```diff
@@ -176,12 +183,12 @@ func (p *processor) PushDefaultEvaluationEvent(user *user.User, featureID string
 		0,
 		p.sourceID,
 		user,
-		&model.Reason{Type: model.ReasonClient},
+		&model.Reason{Type: reason},
 	)
 	encodedEvaluationEvt, err := json.Marshal(evaluationEvt)
 	if err != nil {
 		p.loggers.Errorf(
-			"bucketeer/event: PushDefaultEvaluationEvent failed (err: %v, UserID: %s, featureID: %s)",
+			"bucketeer/event: PushDefaultEvaluationEventWithReason failed (err: %v, UserID: %s, featureID: %s)",
```

Comment:
We could also log the reason for better debugging, as the name is `PushDefaultEvaluationEventWithReason`, WDYT?